packages/react-core/src/components/TextInput/TextInput.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/FormControls/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const inputTypes = [
+  'text',
+  'date',
+  'datetime-local',
```

Comment:
Do we really want to allow the default HTML date time picker?  It is not supported by most browsers and it won't have the PF styles applied.
Same comment goes for the "time", "date" and "month" types.

Reply:
so do we just want to allow only `number, text, password` ?

Reply:
@dmiller9911

Reply:
I don't think it hurts allowing it. This really should be a list of input types that match the spec honestly. I think we could remove the date and datetime-local once we offer a date/time component, but until then we need to allow the consumers to do it if they need to. The alternative is them building their own while referencing class names from core. I would rather avoid that possibility. No consumer should need to turn to core css if using React.

Reply:
I think that since this is a low level component we can be more flexible with this until we get into the more high level components where we can be more specific. We really just want to make sure that we aren't allowing them to pass in `type='radio'` or `type='checkbox'` but I think it'll be good to give consumers the flexibility of being allowed to enter the other input types based off of their needs. If we create our own search component and date/time component then we can revisit removing these.